logger/log.go

```diff
@@ -0,0 +1,65 @@
+package logger
+
+import (
+	"fmt"
+	"runtime"
+	"time"
+
+	"github.com/fatih/color"
+)
+
+//Stores the colors once
+type logger struct {
+	c map[string]*color.Color
+}
+
+var instance *logger
+
+//Creates the Singleton Logger
+func InitLogger() {
+	//TODO Needs to accept a path to output the log to.
+	//TODO Maybe accept a logging level on what to display. That way, someone
+	//can run this on DEBUG mode and have all the DEBUG messages print out.
+	instance = &logger{}
+	instance.c = make(map[string]*color.Color)
+	instance.c["cyan"] = color.New(color.FgCyan)
+	instance.c["yellow"] = color.New(color.FgHiYellow)
+}
+
+//Returns an instance to the logger
+func GetLogger() *logger {
+	return instance
+}
+
+//A wrapper function for fmt.Printf. This function will
+//Look for the caller of the Logf function and automatically
+//print in the following format:
+//<timestamp><caller><Message>
+func (l *logger) Logf(format string, args ...interface{}) string {
```

Comment:
Let's just rename to Printf, because I'd like to add Fatal, Error, and Warning calls as well.